stix2/sources/__init__.py

```diff
@@ -314,175 +310,301 @@ def add_filter(self, filters):
 
-        """
 
+        """
         for filter_id in filter_ids:
             try:
                 if filter_id in self.filters:
                     del self.filters[filter_id]
                     del self.filter_allowed[filter_id]
             except KeyError:
-                # filter id not found in list of filters
-                # attached to the Composite Data Source
+                # filter 'id' not found list of filters attached to Data Source
                 pass
 
         return
 
     def get_filters(self):
-        """return filters attached to Composite Data Source
+        """return copy of all filters currently attached to Data Source
 
-        Args:
+        TODO: make this a property?
 
         Returns:
-            (list): the list of filters currently attached to the Data Source
+            (list): a copy of all the filters(dict) which are attached
+                to Data Source
 
         """
         return copy.deepcopy(list(self.filters.values()))
 
-    def deduplicate(self, stix_obj_list):
-        """deduplicate a list fo STIX objects to a unique set
+    def apply_common_filters(self, stix_objs, query):
+        """evaluates filters against a set of STIX 2.0 objects
 
-        Reduces a set of STIX objects to unique set by looking
-        at 'id' and 'modified' fields - as a unique object version
-        is determined by the combination of those fields
+        Supports only STIX 2.0 common property fields
 
         Args:
-            stix_obj_list (list): list of STIX objects (dicts)
+            stix_objs (list): list of STIX objects to apply the query to
+            query (list): list of filters (combined form complete query)
 
         Returns:
-                (list): unique set of the passed list of STIX objects
+            (list): list of STIX objects that successfully evaluate against
+                the query
+
         """
 
-        unique = []
-        dont_have = False
-        for i in stix_obj_list:
-            dont_have = False
-            for j in unique:
-                for field in STIX_VERSION_FIELDS:
-                    if not i[field] == j[field]:
-                        dont_have = True
-                    break
-            if dont_have:
+        filtered_stix_objs = []
+
+        # evaluate objects against filter
+        for stix_obj in stix_objs:
+            clean = True
+            for filter_ in query:
+
+                # skip filter as filter was identified (when added) as
+                # not a common filter
+                if 'id' in filter_ and self.filter_allowed[filter_['id']] is False:
+                    continue
+
+                # check filter "field" is in STIX object - if cant be applied
+                # due to STIX object, STIX object is discarded (i.e. did not
+                # make it through the filter)
+                if filter_['field'] not in stix_obj.keys():
+                    clean = False
+                    break
```

Comment:
Instead of setting `clean` to `False`, could you just do `continue` if you determine the object shouldn't be added to result? This could simplify the code.